combineRooms/combineRooms.py

```diff
@@ -0,0 +1,320 @@
+
+    @commands.command(aliases=["setroomcap", "src"])
+    @commands.guild_only()
+    @checks.admin_or_permissions(manage_guild=True)
+    async def set_room_capacity(self, ctx, size: int):
+        if size < 2:
+            await ctx.send("No.")
+            return False  
+        combines_cat = await self._save_room_capacity(ctx.guild, size)
+        if self._combines_category(ctx.guild):
+            await ctx.send("Done: Changes will not be applied to rooms that are already up.")
+        else:
+            await ctx.send("Done")
+        return True
+
+    @commands.Cog.listener("on_voice_state_update")
+    async def on_voice_state_update(self, member, before, after):
+        combines_ongoing = await self._combines_category(member.guild)
+
+        # ignore when combines are not ongoing, or when voice activity is within the same room
+        if not combines_ongoing or before.channel == after.channel:
+            return
+        
+        # Room joined:
+        await self._member_joins_voice(member, after.channel)
+        
+        # Room left:
+        await self._member_leaves_voice(member, before.channel) # TODO: consider disconnected case #@me what does that even mean? this structure should cover everything
+
+    async def _start_combines(self, ctx):
+        # Creates combines category and rooms for each tier
+        combines_category = await self._add_combines_category(ctx, self._combines_category_name)
+        await self._save_combine_category(ctx.guild, combines_category)
+
+        if combines_category:
+            await self._add_combines_info_channel(ctx.guild, "Combines Details")
+            for tier in await self.team_manager_cog.tiers(ctx):
+                await self._add_combines_voice(ctx.guild, tier)
+            return True
+        return False
+
+    async def _stop_combines(self, ctx):
+        # remove combines channels, category
+        combines_category = await self._combines_category(ctx.guild)
+        if combines_category:
+            for channel in combines_category.channels:
+                await channel.delete()
+            await combines_category.delete()
+            return True
+        await ctx.send("could not find combine rooms.")
+        return False
+
+    def _get_channel_by_name(self, guild: discord.Guild, name: str):
+        for channel in guild.channels:
+            if channel.name == name:
+                return channel
+    
+    async def _add_combines_category(self, ctx, name: str):
+        category = await self._get_category_by_name(ctx.guild, name)
+        
+        if category:
+            await ctx.send("A category with the name \"{0}\" already exists".format(name))
+            return None
+        
+        league_role = self._get_role_by_name(ctx.guild, "League")
+        overwrites = {
+            ctx.guild.default_role: discord.PermissionOverwrite(view_channel=False, connect=False),
```

Comment:
Is there a reason we don't want everyone to view these channels? I'd think seeing a bunch of people playing may make spectators interested in joining.

Reply:
My thinking was that players who aren't signed up to participate in the next season (Draft or Free Agent) won't benefit from seeing the Combines server. But I could remove this line if you'd like.

Reply:
Updated: Combines may be public or private. Permissions vary based on this state.